lib/importers/tp/importer.rb

```diff
@@ -1,15 +1,16 @@
 module Importers
   module TP
     class Importer
-      def initialize(retriever: Retriever, call_record: CallRecord, saver: Saver)
-        @retriever = retriever.new(config: Rails.configuration.x.tp)
+      def initialize(retriever: Retriever, call_record: CallRecord, saver: Saver, config: Rails.configuration.x.tp)
+        @retriever = retriever.new(config: config)
```

Comment:
It's not a pattern i'm overly-enamoured with, but using a globally scoped configuration object might win out over having to pass this `config` around several layers. Alternatively, why not just make the `Retriever`'s `config` default to `Rails.configuration.x.tp` too? So we don't need to pass it along here...